bonsai/sdk/src/alpha.rs

```diff
@@ -482,11 +485,14 @@ impl Client {
             .send()?;
 
         if !res.status().is_success() {
+            if res.status().as_u16() == 404 {
+                return Err(SdkErr::ReceiptNotFound);
+            }
             let body = res.text()?;
             return Err(SdkErr::InternalServerErr(body));
```

Comment:
I wonder if it could be the case to add a check here to ensure the body is not empty, otherwise we could still present an empty error message to users.

Reply:
I'm not sure this is an issue, sometimes we want to be opaque so as to not reveal what is going on in the backend.